src/networking/backend.ts

```diff
@@ -63,17 +64,24 @@ export class Backend {
       product_id: string;
       email: string;
       presented_offering_identifier: string;
+      offer_id?: string | undefined;
     };
 
+    const requestBody: SubscribeRequestBody = {
+      app_user_id: appUserId,
+      product_id: productId,
+      email: email,
+      presented_offering_identifier: offeringIdentifier,
+    };
+
+    if (purchaseOptionId && purchaseOptionId !== "base_option") {
```

Comment:
Maybe we should move this logic to the backend.

The 'base_option' is a fake one that we return whenever we do not have an offer.

The backend endpoint expects an existing `offer_id` if specified, otherwise it shouldn't be passed.

We can definitely improve this as a second step (since it's not publicly exposed) and make sure that the backend accepts 'base_option' as if the `option_id` was not set.

Reply:
I created [this ticket](https://linear.app/revenuecat/issue/BIL-608/%5Btrials%5D-%5Bbackend%5D-accept-base-plan-in-the-purchase-endpoint-as-offer) to make this change.

Reply:
Shipped!